exporter/opentelemetry-exporter-otlp-proto-grpc/tests/test_otlp_exporter_mixin.py

```diff
@@ -26,20 +26,213 @@
 from google.rpc.error_details_pb2 import (  # pylint: disable=no-name-in-module
     RetryInfo,
 )
-from grpc import Compression
+from grpc import Compression, server
 
-from opentelemetry.exporter.otlp.proto.grpc.exporter import (
-    ExportServiceRequestT,
+from opentelemetry.exporter.otlp.proto.common.trace_encoder import (
+    encode_spans,
+)
+from opentelemetry.exporter.otlp.proto.grpc.exporter import (  # noqa: F401
     InvalidCompressionValueException,
     OTLPExporterMixin,
-    RpcError,
-    SDKDataT,
     StatusCode,
     environ_to_compression,
 )
+from opentelemetry.exporter.otlp.proto.grpc.version import __version__
+from opentelemetry.proto.collector.trace.v1.trace_service_pb2 import (
+    ExportTraceServiceRequest,
+    ExportTraceServiceResponse,
+)
+from opentelemetry.proto.collector.trace.v1.trace_service_pb2_grpc import (
+    TraceServiceServicer,
+    TraceServiceStub,
+    add_TraceServiceServicer_to_server,
+)
+from opentelemetry.sdk.environment_variables import (
+    OTEL_EXPORTER_OTLP_COMPRESSION,
+)
+from opentelemetry.sdk.trace import ReadableSpan, _Span
+from opentelemetry.sdk.trace.export import (
+    SpanExporter,
+    SpanExportResult,
+)
+
+
+# The below tests use this test SpanExporter and Spans, but are testing the
+# underlying behavior in the mixin. A MetricExporter or LogExporter could
+# just as easily be used.
+class OTLPSpanExporterForTesting(
+    SpanExporter,
+    OTLPExporterMixin[
+        ReadableSpan, ExportTraceServiceRequest, SpanExportResult
+    ],
+):
+    _result = SpanExportResult
+    _stub = TraceServiceStub
+
+    def _translate_data(
+        self, data: Sequence[ReadableSpan]
+    ) -> ExportTraceServiceRequest:
+        return encode_spans(data)
+
+    def export(self, spans: Sequence[ReadableSpan]) -> SpanExportResult:
+        return self._export(spans)
+
+    @property
+    def _exporting(self):
+        return "traces"
+
+
+class TraceServiceServicerWithExportParams(TraceServiceServicer):
+    def __init__(
+        self,
+        export_result,
+        optional_export_sleep=None,
+        optional_export_retry_millis=None,
+    ):
+        self.export_result = export_result
+        self.optional_export_sleep = optional_export_sleep
+        self.optional_export_retry_millis = optional_export_retry_millis
+
+    # pylint: disable=invalid-name,unused-argument,no-self-use
```

Comment:
nit: unnecessary? if the self use is recognized by pylint
```suggestion
    # pylint: disable=invalid-name,unused-argument
```

Reply:
Yeah you're right it's not needed. Removed